### PR TITLE
Merge plugin

### PR DESCRIPTION
**Pull request by ArnaudWeyts**
### Related issues

#31 

### Added

* Test cases for a merge plugin
* An implementation of a merge plugin, it won't work yet, as we need access to the `include` param from the request.

Base branch is an integration branch, because we need to refactor the way plugins receive their parameters. The merge plugin should only be used on a response, but it needs access to both the request data and the response data.

### CODE REVIEW

**Comment by rathesDot** [comment]
Should we also add a test that proves that the same entity can be replaced multiple times? Or is that something that can never occur?

**Comment by ArnaudWeyts** [comment]
@rathesDot the test `should merge the data when the parent path is an array` already covers that though, but we can add one for non-array references as well

File: src/plugins/merge/merge.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import _set from 'lodash/fp/set';
+
+import { convertPathToKeys, resolveReferences } from './referenceResolver';
+
+const mergeIncludedReferenceIntoEntity = (entity, included, reference, path) => {
+  const includedEntity = included[reference.type].find(includedEntry => includedEntry.id === reference.id);
+
+  return _set(path, { ...reference, ...includedEntity }, entity);
```
Comment (Kemosabert) [inline src/plugins/merge/merge.js]: Q: I am not that familiar with this `set` function from Lodash, but I looked it up and it seems like the arguments you pass to it, are not in the right order. 

The docs say it should be `(object, path, value)`, but it seems like you have `(path, value, object)`. Am I missing something here?

Comment (ArnaudWeyts) [reply src/plugins/merge/merge.js]: This is the `lodash/fp` version of `set`, which returns a new object, as opposed to mutating it. For some reason the arguments are in a different order here, don't ask 🙄 

https://gist.github.com/jfmengels/6b973b69c491375117dc#_setpath-value-object

Comment (Kemosabert) [reply src/plugins/merge/merge.js]: aha, all good then :)

**Comment by ArnaudWeyts** [comment]
@lowiebenoot yes, definitely will have to put some more work into making it compatible with the other plugins

File: src/plugins/merge/merge.js (hunk 1 above)
Comment (duivvv) [reply src/plugins/merge/merge.js]: > For some reason the arguments are in a different order here, don't ask

I think it's for currying reasons :)

**Pull request merged**